lib/worker-config.js

```diff
@@ -0,0 +1,293 @@
+    }
+    this.configuration = configuration;
+  }
+
+  /**
+   * Evaluate a worker configuration
+   */
+  evaluate(satisfiers) {
+    return this.configuration;
+  }
+
+  workerTypes() {
+    return this._workerTypes.values();
+  }
+}
+
+/**
+ * This class represents a provisioned worker configuration in memory. *
+ *
+ * The constructor parameters are:
+ *  - id: identifier for this worker configuration
+ *  - workerTypes: list of objects {workerType, biddingStrategy, providers}
+ *  - rules: A Ruleset to use with this worker type
+ *  - biddingStrategies: reference to a Map which maps bidding strategy
+ *    identifier to the bidding strategy implmenentation
+ *  - providers: reference to a Map which maps provider identifier to the
+ *    provider implementation
+ */
+class ProvisionedWorkerConfiguration extends WorkerConfiguration {
+  constructor({id, workerTypeConfigurations, rules}) {
+    super({id});
+
+    for (let workerTypeConfiguration of workerTypeConfigurations) {
+      let workerType = workerTypeConfiguration.get('workerType');
+      let providerIds = workerTypeConfiguration.get('providerIds');
+      let biddingStrategyId = workerTypeConfiguration.get('biddingStrategyId');
+
+      if (typeof workerType !== 'string') {
+        this._throw(errors.InvalidWorkerConfiguration, 'worker type name must be string');
+      }
+      this.workerType = workerType;
+
+      if (typeof biddingStrategyId !== 'string') {
+        this._throw(errors.InvalidWorkerConfiguration, 'bidding strategy id must be string');
+      }
+      this.biddingStrategyId = biddingStrategyId;
+
+      if (!Array.isArray(providerIds)) {
+        this._throw(errors.InvalidWorkerConfiguration, 'provider ids must be array');
+      }
+      for (let providerId of providerIds) {
+        if (typeof providerId !== 'string') {
+          this._throw(errors.InvalidWorkerConfiguration, 'provider id must be string');
+        }
+      }
+      this.provider = providerIds;
+    }
+
+    this.workerTypeConfigurations = workerTypeConfigurations;
+
+    if (typeof rules !== 'object') {
+      this._throw(errors.InvalidWorkerConfiguration, 'rules must be provided');
+    }
+    this.rules = rules;
+  }
+
+  /**
+   * Evaluate a WorkerConfiguration's rules based on a set of satisfiers and a
+   * map of providers.  The satisfiers must contain a provisionerId which is an
+   * entry in the providers map
+   */
+  evaluate({providers, satisfiers}) {
+    if (typeof satisfiers !== 'object') {
+      this._throw(errors.InvalidSatisfiers, 'invalid satisfiers provided'); 
+    }
+    let providerId = satisfiers.providerId;
+    let provider = providers.get(providerId);
+
+    if (!provider) {
+      this._throw(errors.InvalidProvider, `unknown provider: ${providerId}`);
+    }
+
+    let requiredSatisfiers = this.rules.requiredSatisfiers().sort();
+    let providedSatisfiers = provider.providedSatisfiers();
+    if (!Array.isArray(providedSatisfiers)) {
```

Comment:
I've gotten rid of this assertion in the code, but you're right that it would've been unnecessary once we had a real Fake Provider (:laughing:) to test against, but my concern is that the fake provider in my tests were sort of... simple, so I wanted to be overzealous up front.